src/util/CancellationHandle.h

```diff
@@ -167,27 +183,33 @@ class CancellationHandle {
       const InvocableWithConvertibleReturnType<
           std::string_view, ArgTypes...> auto& detailSupplier,
       ArgTypes&&... argTypes) {
-    auto state = cancellationState_.load(std::memory_order_relaxed);
-    if (state == CancellationState::NOT_CANCELLED) [[likely]] {
-      return;
-    }
-    if constexpr (WatchDogEnabled) {
-      if (!detail::isCancelled(state)) {
-        pleaseWatchDog(state, detailSupplier, AD_FWD(argTypes)...);
+    if constexpr (CancellationEnabled) {
```

Comment:
Don't you get warnings about the arguments being unused in the cancellation disabled case? If so you have to do any of the `[[maybe_unused]]` or `(void)` tricks.